Plutarch/Api/V1/Tuple.hs

```diff
@@ -0,0 +1,25 @@
+module Plutarch.Api.V1.Tuple (PTuple, ptuple, ptupleFromBuiltin, pbuiltinPairFromTuple) where
+
+import Plutarch.Builtin (pconstrBuiltin, pforgetData)
+import Plutarch.Prelude
+import Plutarch.Unsafe (punsafeCoerce)
+
+type PTuple a b =
+  PDataSum
```

Comment:
Why not `PDataRecord`? Seems unnecessary.

Reply:
I'm also curious why `PTuple` was originally defined as a `PDataSum`. It could just be a `newtype` to a `PDataRecord`. It can't be `PDataRecord` alone though. That's just a builtin list, `PTuple` is a `Constr` data.

Reply:
No, I mean, why represent it as a `Constr`?

Reply:
Well we have to - that's how the ledger api does it. Tilde discovered this originally.

Reply:
Ah I see, I didn't realise we were using it there.